### ISSUE TITLE
Move top level task list into another task list

### ISSUE THREAD

**Issue by janvlug** [nextcloud/tasks#2700]
### Is your feature request related to a problem? Please describe.

I would like to be able to drag a top level task list (as shown at the left of the tasks web page) into another top level task list, so that it will appear there as a sub task.

My use case: In this way I could clean up the amount of top level task lists.

**Comment by janvlug** [comment]
@raimund-schluessler: Yes you understand me correctly. And the moved tasks should be in the other list under a top level task with the same name as the task list that is being moved.

Maybe it is a bit more clear if I explain it by using screenshots and URL's:
Imagine that there are to task list:

![Image](https://github.com/user-attachments/assets/15e507c3-3c9c-489b-a562-dc065af990d1)

- `https://example.com/index.php/apps/tasks/calendars/tasklistA` (referred to as `tasklistA`)
- `https://example.com/index.php/apps/tasks/calendars/tasklistB` (referred to as `tasklistB`)

Now, I would like to be able to move `tasklistB` under `tasklistA`. The URL for `tasklistB` as mentioned above would disappear, and `tasklistB` would appear under `tasklistA` as this:

![Image](https://github.com/user-attachments/assets/7321514b-8ecb-4feb-8de9-c00a148eea07)

The URL's woud be something like this:
- `https://example.com/index.php/apps/tasks/calendars/tasklistA`
- `https://example.com/index.php/apps/tasks/calendars/tasklistA/tasks/<uuid of new task representing tasklistB>.ics`

**Comment by dateifabrik** [comment]
Might be nice to move a subtask to another main task too.
